Tests/SwiftSyntaxTestSupportTest/SourceEditsTestUtilsTest.swift

```diff
@@ -0,0 +1,54 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import XCTest
+import SwiftSyntax
+import _SwiftSyntaxTestSupport
+
+public class SourceEditsUtilTest: XCTestCase {
+  public func testDiffOfTwoStringsSimple() throws {
+    let s1 = "struct A { func f() {"
+    let s2 = "struct AA { func f() {"
+
+    let diffs = getConcurrentEdits(from: s1, to: s2)
+    XCTAssertEqual(diffs.edits.count, 1)
+
+    let firstDiff = try XCTUnwrap(diffs.edits.first)
+    XCTAssertEqual(firstDiff, SourceEdit(offset: 8, length: 0, replacementLength: 1))
+  }
+
+  public func testDiffOfTwoSameStrings() {
+    let s1 = "0123456"
+
+    let diffs = getConcurrentEdits(from: s1, to: s1)
+    XCTAssert(diffs.edits.isEmpty)
+  }
+
+  public func testDiffOfTwoStrings() {
+    let s1 = "0123456"
+    let s2 = "x12456yz"
+
+    let diffs = getConcurrentEdits(from: s1, to: s2)
+
+    let expectedDiffs: [SourceEdit] = [
+      SourceEdit(offset: 0, length: 1, replacementLength: 1),
+      SourceEdit(offset: 3, length: 1, replacementLength: 0),
+      SourceEdit(offset: 7, length: 0, replacementLength: 2),
+    ]
+
+    XCTAssertEqual(diffs.edits, expectedDiffs)
+
+    let s3 = applyEdits(expectedDiffs, concurrent: true, to: s1)
+
+    XCTAssertEqual(s3, "?12456??")
+  }
```

Comment:
Could you add one more test case that has an insertion at the start and an insertion at the end. That would make sure that `Collection.difference` actually contains sequential edits instead of concurrent edits.

```swift
let base = "0123"
let new = "a0123b"
```

Reply:
Sorry if I wasn’t clear here. What I was looking for is a test case just like the one you have in `testDiffOfTwoStrings` (which I think has a very good test structure), just with the inputs I gave you. So basically:

```swift
  public func testStringDifferenceReturnSequentialEdits() throws {
    let base = "0123"
    let new = "a0123bc"

    let diffs = getConcurrentEdits(from: base, to: new)
    XCTAssertEqual(diffs.edits, [
        SourceEdit(offset: 0, length: 0, replacementLength: 1),
        SourceEdit(offset: 4, length: 0, replacementLength: 2)
      ])

    let appliedDiffsBase = applyEdits(diffs.edits, concurrent: true, to: base)

    XCTAssertEqual(appliedDiffsBase, "?0123??")
  }
```

Which seems to pass, so that’s good ✅

Reply:
I think this `replacementLength > 0` case is covered in `testDiffOfTwoStrings`. Do you think we should keep the current version of `testStringDifferenceReturnSequentialEdits` (which makes sure `String.difference` correctly produce a string of sequential edits)?